R/workability.R

```diff
@@ -1,257 +1,257 @@
-#' ind_workability(D_WO = c(0.15,0.6,0.9),B_LU_BRP = c(256,1019,1019))
-#'  
-#' @return 
-#' The evaluated score for the soil function to allow the soil to be managed by agricultural activities. A numeric value between 0 and 1.
-#' 
-#' @export
-ind_workability <- function(D_WO, B_LU_BRP) {
-  
-  # add visual bindings
-  id = arg.length = crop_code = crop_season = rsl = . = NULL
-  
-  # Load in the datasets
-  crops.obic <- as.data.table(OBIC::crops.obic)
-  setkey(crops.obic, crop_code)
-  
-  # length of inputs
-  arg.length <- max(length(D_WO), length(B_LU_BRP))
-  
-  # Check inputs
-  checkmate::assert_numeric(D_WO, lower = 0, upper = 1, any.missing = FALSE)
-  checkmate::assert_numeric(B_LU_BRP, any.missing = FALSE, min.len = 1, len = arg.length)
-  checkmate::assert_subset(B_LU_BRP, choices = unique(crops.obic$crop_code), empty.ok = FALSE)
-  
-  # Form data table
-  dt <- data.table(id = 1:arg.length,
-                   rsl = D_WO,
-                   B_LU_BRP = B_LU_BRP)
-  
-  # Merge crop_season into data table
-  dt <- merge.data.table(dt, crops.obic[,.(crop_code, crop_season)], by.x = 'B_LU_BRP', by.y = 'crop_code')
-  
-  # evaluate relative season length
-  dt[!grepl('^beweid', crop_season),score := evaluate_logistic(x = rsl, b = 15, x0 = 0.75, v = 1, increasing = TRUE)]
-  dt[grepl('^beweid', crop_season),score := evaluate_logistic(x = rsl, b = 9, x0 = 0.5, v = 1, increasing = TRUE)]
-  
-  # overwrite score when rsl = 1
-  dt[rsl == 1, score := 1]
-  
-  # setorder
-  setorder(dt, id)
-  
-  # Return score
-  score <- dt[,score]
-  
-  return(score)
-}
+#' Calculate indicator for workability 
+#'
+#' This function calculates the workability of soils, given as a value of relative season length between 0 and 1. 
+#' A relative season length of 1 indicates that the water table is sufficiently low for the soil to be workable for the entire growing season required by the crop.
+#' The required ground water table for workability is determined by soil type and soil properties. Hydrological variables determine the groundwater table for each day of the year.
+#' The option calcyieldloss allows for calculation of yield loss based on the relative season length, differentiating in yield loss between six groups of crops
+#' Based on Huinink (2018)
+#' 
+#' @param A_CLAY_MI (numeric) The clay content of the soil (\%)
+#' @param A_SILT_MI (numeric) The silt content of the soil (\%)
+#' @param B_LU_BRP (numeric) The crop code from the BRP
+#' @param B_SOILTYPE_AGR (character) The agricultural type of soil
+#' @param B_GWL_GLG (numeric) The lowest groundwater level averaged over the most dry periods in 8 years in cm below ground level
+#' @param B_GWL_GHG (numeric) The highest groundwater level averaged over the most wet periods in 8 years in cm below ground level
+#' @param B_GWL_ZCRIT  (numeric) The distance between ground level and groundwater level at which the groundwater can supply the soil surface with 2mm water per day (in cm)

```

Comment:
B_GWL_ZCRIT should be defined as
"Distance between groundwater table and root zone (30 cm -mv) for delivering 2 mm/day"
As is the description of the input data we use.

This has consequences for the changes in line 117. For example, if the spring_workindepth == 30, the Zcrit (to the root zone 0-30cm) already accounts for this working depth and no addition or substraction is needed.

Reply:
I pushed another script now that resolves this by firstly chaing the definition in line 15 and then changing the calculation we discussed to Zcrit + springdepth - 30. This way if the working depth is less than the rootzone the depth at which the capilary rise is lower than the evaporation is slightly decreased, if the working depth is larger than the rootzone this depth increases. Good thing is that with this definition the change in depth is smaller making the assumption the soil profile remains the same more reasonable.

I'm not sure if committing went well, please let me know. I can't see how I can ask you to review.